### PR TITLE
Improve usability of refund

### LINKED ISSUE #683: [1.0 XMR] Provide better feedback when cancel/refund transaction publish fails by parsing electrum error
As discovered in https://github.com/comit-network/xmr-btc-swap/issues/674, users may try to refund a swap even though that is no longer possible because Alice already punished.

We should catch this case by parsing the error message returned from electrum and display something more informative to the user.

Also, we need to advance the internal state to correctly reflect what happened on the blockchain, otherwise `swap list` returns misleading information.

### PR DESCRIPTION
**Pull request by thenhnn**
User can try to refund even if it is no longer possible. We will show plain text error rather than ugly JSON error, see #674 for more info.

Fixes #683

### CODE REVIEW

**Comment by thenhnn** [comment]
I'm not sure about `fn cancel` because it does error handling in proper way (as i understand), but logs it to tracing, rather than showing it to user.

**Comment by delta1** [comment]
Please work on an accompanying test, you could use this one as a starting point 

https://github.com/comit-network/xmr-btc-swap/blob/master/swap/tests/alice_manually_punishes_after_bob_dead.rs

File: swap/src/cli/cancel_and_refund.rs [hunk 1]
```diff
@@ -105,7 +105,23 @@ pub async fn refund(
...
+        Ok(_) => (),
+        Err(refund_error) => {
+            if let Some(refund_error) = refund_error.downcast_ref::<BtcPunishedWhileRefundError>() {
+                let state = BobState::BtcPunished {
+                    tx_lock_id: state6.tx_lock_id(),
+                };
+                db.insert_latest_state(swap_id, state.clone().into())
+                    .await?;
+
+                println!("{}", refund_error);
```
Comment (binarybaron) [inline swap/src/cli/cancel_and_refund.rs:119]: Please always use tracing for printing logs and add a message for some context.

**Comment by thenhnn** [comment]
sorry, can you unassign me? i'm a bit busy and i'm not sure if i can make this.

**Comment by binarybaron** [comment]
Closing this in favour of https://github.com/comit-network/xmr-btc-swap/pull/1668

**Pull request closed without merging**